packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -0,0 +1,89 @@
+import { Payment } from "@models/payment"
+import { Cart } from "@models/cart"
+import { PaymentSession, PaymentSessionStatus } from "@models/payment-session"
+import { Customer } from "@models/customer"
+import { BaseService } from "medusa-interfaces"
+
+export type Data = Record<string, unknown>
+export type PaymentData = Data
+export type PaymentSessionData = Data
+
+export interface PaymentService {
+  getIdentifier(): string
+
+  onCreatePaymentSession(cart: Cart): Promise<PaymentSessionData>
+
+  onCreatePayment(paymentSession: PaymentSession): Promise<PaymentData>
+
+  retrievePayment(cart: Cart): Promise<Payment>
+
+  updatePayment(
+    paymentSessionData: PaymentSessionData,
+    cart: Cart
+  ): Promise<PaymentSessionData>
+
+  authorizePayment(
```

Comment:
Should we name this one `onAuthorizePayment`?

Reply:
it really is the action to authorizing the payment with the external provider, it is not just building some data on authorized action so I think keeping the name is fine

Reply:
By this standard, however, I feel that `onCreatePaymentSession` is also being named incorrectly as `onCreatePaymentSession` in most cases will be an actual call to a 3rd party API to "initialize" a session, e.g. Stripe creates a PaymentIntent, Klarna creates a Checkout instance etc. 

Maybe let's circle back once again and look at what we are trying to do here:
- Choose names that clearly indicate to a plugin author what the logic in the function should do
- Strengthen the type safety and versatility of the interface

To the first point, I think both `onAuthorizePayment` and `authorizePayment` are equally good at explaining what happens - the `on`-prefix could to a certain extent indicate more clearly that the method is being called as part of some other logic orchestration (namely the Cart service, when it is trying transforming or verifying that a payment is authorized) although I don't think that in and of itself makes it really clear. Another thing that I think is quite important in the interface is that we are consistent so if we name one method `onXYZ` then I think we should name the rest `onABC`.

With all this being said maybe we should go back to dropping `on` from all function names and just make sure that they are correctly explaining what they are creating or what they are doing: e.g. `createPaymentSession`, `authorizePaymentSession`, `updatePaymentSession`, `createPayment`, `capturePayment`, etc.